### PR TITLE
chore: Add the kubernetes architecture outline to the CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by MOZGIII**
Part of #6396.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -910,6 +910,23 @@ or
...
+Kubernetes integration architecture is largely inspired by
+the [RFC 2221](rfcs/2020-04-04-2221-kubernetes-integration.md), so this
+is a concise outline of the effective design, rather than a deep dive into
+the concepts.
+
+With `kubernetes_logs` source, Vector connects to the Kubernetes API doing
+a streaming watch request over the `Pod`s executing on the same `Node` that
+Vector itself runs at. Once Vector gets the list of all the `Pod`s that are
+running on the `Node`, it starts collecting logs for the logs files corresponding to each of the `Pod`. Only plaintext (as in non-gzipped) files
+are taken into consideration.
```
Comment (bruceg) [inline CONTRIBUTING.md:924]: These two lines should get word wrapped the same as the others.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -910,6 +910,23 @@ or
 QUICK_BUILD=true CONTAINER_IMAGE_REPO=<your name>/vector-test make test-e2e-kubernetes
 ```
 
+#### Kubernetes Architecture
```
Comment (bruceg) [inline CONTRIBUTING.md:913]: Does this need to be added to the index at the top of the file?

**Review by bruceg — COMMENTED** [review]
This could possibly use some paths to reference where the various parts are found, otherwise looks OK.

File: CONTRIBUTING.md (hunk 2 above)
Comment (MOZGIII) [reply CONTRIBUTING.md:913]: There's a MarkdownTOC annotation. Do you happen to know how do I trigger the regeneration? I used a VS Code plugin for things like these before, but it's broken atm.

Comment (binarylogic) [reply CONTRIBUTING.md:913]: It's a sublime text plugin :(. I would just manually add it for now.

**Comment by binarylogic** [comment]
Thanks @MOZGIII

**Pull request merged**